src/main/java/com/salesforce/dataloader/controller/Controller.java

```diff
@@ -234,43 +243,88 @@ public synchronized boolean saveConfig() {
+     *  False if directory was failed to create
      */
-    private void copyFile(File file, File destFile) {
-        BufferedInputStream in = null;
-        BufferedOutputStream out = null;
-        try {
-
-            // reading and writing 8 KB at a time
-            int bufferSize = 8 * 1024;
+    private static boolean createDir(File dirPath) {
+        boolean isSuccessful = true;
+        if (!dirPath.exists() || !dirPath.isDirectory()) {
+            isSuccessful = dirPath.mkdirs();
+            if (isSuccessful) {
+                logger.info("Created config directory: " + dirPath);
+            } else {
+                logger.info("Unable to create config directory: " + dirPath);
+            }
+        } else {
+            logger.info("Config directory already exists: " + dirPath);
+        }
+        return isSuccessful;
+    }
 
-            byte[] buf = new byte[bufferSize];
+    /**
+     * Returns current user's Dataloader configuration directory
+     *  ie) For Windows - C:\Users\{user}\AppData\Local\salesforce.com\Data Loader {version}\conf
+     *      For Mac - /Users/{user}/Library/Preferences/salesforce.com/Data Loader {version}/conf
+     *      For *nix - ~/.salesforce.com/Data Loader {version}/conf
+     *
+     * @return
+     *  Current user's Dataloader configuration directory
+     */
+    private static File getUserConfigDir() {
+        File dir;
 
-            in = new BufferedInputStream(new FileInputStream(file), bufferSize);
-            int r = 0;
+        switch (OS_TYPE) {
+            case WINDOWS: {
+                dir = Paths.get(System.getProperty("user.home"), "AppData/Local", APP_VENDOR, getProductName(), CONFIG_DIR).toFile();
+                break;
+            }
+            case MACOSX: {
+                dir = Paths.get(System.getProperty("user.home"), "Library/Preferences", APP_VENDOR, getProductName(), CONFIG_DIR).toFile();
+                break;
+            }
+            default:
+            case LINUX: {
+                dir = Paths.get(System.getProperty("user.home"), "." + APP_VENDOR, getProductName(), CONFIG_DIR).toFile();
+                break;
+            }
 
-            out = new BufferedOutputStream(new FileOutputStream(destFile), bufferSize);
+        }
+        return dir;
+    }
 
-            while ((r = in.read(buf, 0, buf.length)) != -1) {
-                out.write(buf, 0, r);
+    /**
+     * Returns default Dataloader configuration directory
+     * ie) For Windows - C:\Program Files (x86)\salesforce.com\Data Loader\conf
+     *     For Mac - /Applications/Data Loader.app/Contents/Resources/conf
+     *     For *nix - {where DL is installed}/conf
+     *
+     * @return
+     *  Default Dataloader configuration directory
+     */
+    private static File getDefaultConfigDir() {
+        File dir;
+        switch (OS_TYPE) {
+            case WINDOWS: {
+                dir = new File(ProgramDirectoryUtil.getProgramDirectory(), "conf");
```

Comment:
The result of `getProgramDirectory` is not guaranteed to be an install directory (that has a `conf` subdir). Notably, it could be the working directory, if the app is not being run from a .jar file.

If getting the install location is intended to be the result of the function, it should be changed to ensure that behavior.

Reply:
No longer using `getProgramDirectory`